### PR TITLE
feat(client): Unify donation loading state

### PR DESCRIPTION
**Pull request by ahmaxed**
Checklist:

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

There is a lot of room for improvement here, but this pr tackles the following optimizations:
- Partially unified the loading state for the buttons
  (there is an internal loading state for Paypal that will be tackled in upcoming PRs)
- Made the donation buttons section reusable
- extracted some grid and unused elements out of the donateForm

### CODE REVIEW

**Comment by ahmadabdolsaheb** [comment]
the main question is, which one of the following scenarios offer a higher perceived performance  and a better user experience:
1)load each button individually (paypal loads faster and some times google/apple pay is not supported)
2)show a spinner until both load (or until one loads and we know the other is not supported)

**Comment by ShaunSHamilton** [comment]
> higher perceived performance

What does Lighthouse say about the content/interactivity?

> a better user experience

Personally, one where all buttons appear at same time (**2**), because as I am reading the description of why/what I am about to do (donate), I do not want my eyes distracted 2-3 times as the buttons appear.

**Comment by ahmadabdolsaheb** [comment]
@ShaunSHamilton it is hard to say because it would depend on wether the browser that visits our website has google pay enabled or not. 

Since the loader is going to be on the modal as well were there is not much read, it might effect the perceived performance. It is hard to tell if it matter much donation wise, but I think it is a good idea to unify the states at this point and optimize it with A/B testing in upcoming prs.

**Comment by ShaunSHamilton** [comment]
> Since the loader is going to be on the modal 

Is there a specific reason we do not just link to `/donate` on the modal? (no pay buttons)

**Comment by ahmadabdolsaheb** [comment]
> > Since the loader is going to be on the modal
> 
> Is there a specific reason we do not just link to `/donate` on the modal? (no pay buttons)

The longer the user journey the less likely it will be completed. Although I would love to A/B test that to see if the users are affected. That way we can clean up a lot of the components.

**Review by moT01 — DISMISSED** [review]
Well, I didn't see any problems with this. LGTM 🎉

**Review by ShaunSHamilton — DISMISSED** [review]
LGTM 👍 . Discussed some clean-up (unused stuff) with Ahmad, and he will do that in another open PR of his.

File: client/src/components/Donation/DonateForm.tsx [hunk 1]
```diff
@@ -242,88 +247,64 @@ class DonateForm extends Component<DonateFormProps, DonateFormState> {
...
+    return <DonateCompletion {...props} />;
   }
 
-  renderDonationOptions() {
+  renderButtonGroup() {
+    const { donationAmount, donationDuration } = this.state;
     const {
+      donationFormState: {
+        loading: { stripe, paypal }
+      },
```
Comment (ojeytonwilliams) [inline client/src/components/Donation/DonateForm.tsx:281]:
```suggestion
      donationFormState: { loading },
```
If we pull out the booleans here, `paymentButtonsLoading` doesn't make much sense in isolation.

File: client/src/components/Donation/DonateForm.tsx [hunk 2]
```diff
@@ -242,88 +247,64 @@ class DonateForm extends Component<DonateFormProps, DonateFormState> {
...
       addDonation,
-      t,
       defaultTheme,
-      theme
+      theme,
+      t,
+      isMinimalForm
     } = this.props;
-    const { donationAmount, donationDuration } = this.state;
+    const paymentButtonsLoading = stripe && paypal;
```
Comment (ojeytonwilliams) [inline client/src/components/Donation/DonateForm.tsx:289]:
```suggestion
    const paymentButtonsLoading = loading.stripe && loading.paypal;
```
With this change, the nature of `stripe` and `paypal` are obvious.

File: client/src/components/Donation/DonateForm.tsx [hunk 3]
```diff
@@ -334,75 +315,23 @@ class DonateForm extends Component<DonateFormProps, DonateFormState> {
             addDonation={addDonation}
             donationAmount={donationAmount}
             donationDuration={donationDuration}
+            handlePaymentButtonLoad={this.handlePaymentButtonLoad}
             handleProcessing={handleProcessing}
-            isSubscription={isOneTime ? false : true}
+            isPaypalLoading={paypal}
```
Comment (ojeytonwilliams) [inline client/src/components/Donation/DonateForm.tsx:320]:
```suggestion
            isPaypalLoading={loading.paypal}
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Just some nitpicks.

**Review by ojeytonwilliams — APPROVED** [review]

**Review by ShaunSHamilton — APPROVED** [review]
This still LGTM 👍

**Review by moT01 — APPROVED** [review]
🚀

**Pull request merged**